R/dependecies.R

```diff
@@ -0,0 +1,510 @@
+          if (matches_github[i]){
+            name <- gsub("git::.*github\\.com/", "", name)
+            name <- gsub("\\.git", "", name)
+          }
+          subname <- gsub("^.*/", "", as.character(remote_dependencies[i,1]))
+          subname <- gsub("@.*", "", subname)
+          subrepo <- gsub("@.*", "", as.character(remote_dependencies[i,1]))
+          pkg <- list(name = subname
+            , repo = subrepo
+            , version = as.character(remote_dependencies[i,3])
+            , remote = "github")
+          if (grepl("@", name)) {
+            pkg$ref <- gsub(".*@", "", name)
+          }
+          pkg
+        })
+    }
+  }
+  if (length(remote_list) != 0) {
+    non_remote_names <- vapply(non_remote_list, function(pkg) pkg$name, character(1))
+    remote_names <- vapply(remote_list, function(pkg) pkg$name, character(1))
+    if (any(non_remote_names %in% remote_names)) {
+      non_remote_list <- non_remote_list[!non_remote_names %in% remote_names]
+    }
+  }
+  c(non_remote_list, remote_list)
+}
+
+download_package <- function(package) {
+  UseMethod("download_package")
+}
+
+#' Download CRAN package, either current or older version
+download_package.CRAN <- function(package) {
+  remote_version <- get_available_cran_version(package)
+  name <- package$name
+  version <- package$version
+  repo = "http://cran.r-project.org"
+  if (package$is_dependency_package) {
+    version <- NA
+  }
+
+  ###Some packages are available in archive only
+  if (is.na(remote_version)) {
+    archive_addition <- paste0("Archive/", name, "/")
+    url <- paste0(repo, "/src/contrib/", archive_addition)
+    filenames <- RCurl::getURL(url, ftp.use.epsv = FALSE, dirlistonly = TRUE)
+    filenames <- paste(url, strsplit(filenames, "\r*\n")[[1]], sep = "")
+    filenames <- filenames[grepl(paste0(name, "_[0-9\\.\\-]+\\.tar\\.gz"),filenames)]
+    expr <- gregexpr(paste0(name, "_([0-9\\.\\-]+)\\.tar\\.gz"), filenames)
+    filenames <- vapply(regmatches(filenames, expr), function(match) match[1], character(1))
+    archived_version <- gsub("(.*_)([0-9\\.\\-]+)(\\.tar\\.gz)", "\\2", filenames[length(filenames)])
+    if (is.na(version) || package_version(archived_version) > package_version(as.character(version))) {
+      version <- archived_version
+    }
+  } else{
+    # Simply download latest if version happens to be the latest available on CRAN.
+    remote_version <- as.character(remote_version)
+    if (is.na(version) || package_version(remote_version) == package_version(version)) {
+      version <- remote_version
+      archive_addition <- ""
+    } else{
+      archive_addition <- paste0("Archive/", name, "/")
+    }
+  }
+
+  from <- paste0(repo, "/src/contrib/", archive_addition, name, "_", version, ".tar.gz")
+  pkg_tarball <- tempfile(fileext = ".tar.gz")
+  out <- suppressWarnings(tryCatch(
+    download.file(url = from, destfile = pkg_tarball, quiet = notTRUE(getOption('lockbox.verbose')))
+    , error = function(e) e))
```

Comment:
Just use `try` instead of `tryCatch` if you're not going to do anything with the error.